### PR TITLE
add nil check

### PR DESCRIPTION
**Pull request by mcjcloud**
Fix DID() method

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: x/registry/types/did.go [hunk 1]
```diff
@@ -117,9 +121,9 @@ func stringifyVerificationRelationships(auth did.VerificationRelationships) ([]s
 	return res, nil
 }
 
-func (d *DIDDocument) DID() did.DID {
+func (d *DIDDocument) DID() *did.DID {
 	id, _ := did.ParseDID(d.Id)
```
Comment (SCKelemen) [inline x/registry/types/did.go:125]: off-topic, but DIDDocument.Id should probably be DIDDocument.ID to follow not only Go convention, but to be internally consistent as well.
```suggestion
	id, _ := did.ParseDID(d.ID)
```

**Review by SCKelemen — APPROVED** [review]

File: x/registry/keeper/msg_server_who_is.go [hunk 2]
```diff
@@ -95,7 +95,12 @@ func (k msgServer) UpdateWhoIs(goCtx context.Context, msg *types.MsgUpdateWhoIs)
 	for k, v := range msg.GetMetadata() {
 		val.Metadata[k] = v
 	}
-	controllerId, err := doc.GetController(val.DidDocument.DID())
+
+	d := val.DidDocument.DID()
```
Comment (SCKelemen) [inline x/registry/keeper/msg_server_who_is.go:99]: Why id it DidDocument here but DIDDocument further down?

**Review by 6missedcalls — APPROVED** [review]
LGTM

File: x/registry/types/did.go (hunk 1 above)
Comment (mcjcloud) [reply x/registry/types/did.go:125]: We have a lot of inconsistencies and code cleanup to do after we get our initial launch done. Can you file a bug? Same for the below comment.

**Pull request merged**